sda/router.go

```diff
@@ -0,0 +1,550 @@
+package sda
+
+import (
+	"errors"
+	"reflect"
+	"strconv"
+	"strings"
+	"sync"
+
+	"github.com/dedis/cothority/log"
+	"github.com/dedis/cothority/network"
+	"github.com/dedis/crypto/abstract"
+	"golang.org/x/net/context"
+)
+
+// Router is an abstraction to represent the bridge between the communication
+// layer (network, channels etc) and the logical/processing layer (overlay &
+// protocols, services etc). It is a duplex communication link (send/receive)
+// from/to other router of the same type.
+// Typically, for deployement you would use a tcpRouter so it opens tcp ports
+// and communicate through tcp connections. For testing, there is a localRouter
+// which pass all messages through channels going to other localRouter.
+// For the Router to dispatch messages to your struct, you need to register a
+// `Processor` (see the `Dispatcher` interface in processor.go).
+type Router interface {
+	// Run will start the Router:
+	//  * Accepting new connections
+	//  * Dispatching  incoming messages
+	// It is a blocking call which will return until Close() is called.
```

Comment:
"It will block until Close()" is called or "It is a blocking call which *won't* return until Close() is called."